packages/typespec-azure-resource-manager/lib/Legacy/operations.tsp

```diff
@@ -0,0 +1,272 @@
+
+    /**
+     * @dev Delete a resource synchronously
+     * @template Resource The resource being deleted
+     * @template Parameters Optional. Additional parameters after the path parameters
+     * @template Response Optional. The success response(s) for the delete operation
+     */
+    @autoRoute
+    @doc("Delete a {name}", Resource)
+    @delete
+    @deletesResource(Resource)
+    @armResourceDelete(Resource)
+    DeleteSync<
+      Resource extends Azure.ResourceManager.CommonTypes.Resource,
+      Parameters extends {} = {},
+      Response extends {} = ArmDeletedResponse | ArmDeletedNoContentResponse
+    >(...ParentParameters, ...ResourceTypeParameter, ...Parameters): Response | ErrorType;
+
+    /**
+     * @dev Get a resource
+     * @template Resource The resource being read
+     * @template Parameters Optional. Additional parameters after the path parameters
+     * @template Response Optional. The success response for a get operation.
+     */
+    @autoRoute
+    @doc("Get a {name}", Resource)
+    @get
+    @readsResource(Resource)
+    @armResourceRead(Resource)
+    Read<
+      Resource extends Azure.ResourceManager.CommonTypes.Resource,
+      Parameters extends {} = {},
+      Response extends {} = ArmResponse<Resource>
+    >(...ParentParameters, ...ResourceTypeParameter, ...Parameters): Response | ErrorType;
+
+    /**
+     * @dev List a resource
+     * @template Resource The resource being listed
+     * @template Parameters Optional. Additional parameters after the path parameters
+     * @template Response Optional. The response returned by the list
+     */
+    @autoRoute
+    @doc("List {name} resources", Resource)
+    @get
+    @listsResource(Resource)
+    @segmentOf(Resource)
+    @armResourceList(Resource)
+    List<
+      Resource extends Azure.ResourceManager.CommonTypes.Resource,
+      Parameters extends {} = {},
+      Response extends {} = ArmResponse<ResourceListResult<Resource>>
+    >(...ParentParameters, ...Parameters): Response | ErrorType;
+
+    /**
+     * A synchronous resource action.
+     * @template Resource The resource being acted upon
+     * @template Request The request model for the action
+     * @template Response The response model for the action
+     * @template Parameters Optional. Additional parameters after the path parameters
+     */
+    #suppress "@azure-tools/typespec-azure-core/no-private-usage"
+    @autoRoute
+    @armResourceAction(Resource)
+    @Private.enforceConstraint(Resource, Foundations.Resource)
+    @post
+    @returnsDoc("Azure operation completed successfully.")
+    ActionSync<
+      Resource extends Foundations.Resource,
+      Request extends TypeSpec.Reflection.Model | void,
+      Response extends TypeSpec.Reflection.Model | void,
+      Parameters extends {} = {}
+    >(
+      ...ParentParameters,
+      ...ResourceTypeParameter,
+      ...Parameters,
+
+      @doc("The content of the action request")
+      @bodyRoot
+      body: Request,
```

Comment:
I think we want to add two add an additional parameter for every one of these operations that has a request body, like:

```tsp
RequestBody extends {} = {
  @doc("The content of the action request")
      @bodyRoot
      body: Request
},
```

This would allow optional request bodies, which are not standard, but occur in legacy apis

Reply:
It also allows *renaming* the body parameter without decorators.